sw/snRuntime/src/alloc_v2.h

```diff
@@ -83,7 +83,7 @@ inline void *snrt_l1_alloc_compute_core_local(size_t size,
                                               const size_t alignment) {
     snrt_l1_allocator_v2()->next =
         ALIGN_UP(snrt_l1_allocator_v2()->next, alignment);
-    void *retval = snrt_l1_next_v2() + size * snrt_cluster_core_idx();
+    void *retval = ((uint8_t*) snrt_l1_next_v2()) + size * snrt_cluster_core_idx();
```

Comment:
Perhaps it could be even better to just return `uintptr_t` types in the alloc functions, so the arithmetic can be performed on that type, and then cast (only once) to whatever pointer one may need. What do you think? Would that work for your use case?

Reply:
I don't have strong opinion but it will go against the "malloc" convention

```
     void *malloc(size_t size);
```